extensions/ql-vscode/src/abstract-webview.ts

```diff
@@ -101,7 +122,7 @@ export abstract class AbstractWebview<ToMessage extends WebviewMessage, FromMess
     );
   }
 
-  protected abstract getPanelConfig(): WebviewPanelConfig;
+  protected abstract getPanelConfig(): WebviewPanelConfig | Promise<WebviewPanelConfig>;
```

Comment:
I suppose this works because calling `await` on something that's not a promise is a no-op, so you can safely call `await` on either of these return types and end up with the panel config object.

I'm not 100% sure the flexibility of this is worth the mental complexity, but it looks like it works so I'm not objecting.